src/lib.rs

```diff
@@ -30,6 +31,45 @@ mod nodes;
 // TODO(jmmv): Make configurable via a flag and store inside SandboxFS.
 pub const TTL: Timespec = Timespec { sec: 60, nsec: 0 };
 
+/// An error indicating that a path has to be absolute but isn't.
+#[derive(Debug, Fail)]
+#[fail(display = "path {:?} is not absolute", path)]
+pub struct PathNotAbsoluteError {
+    /// The path that caused this error.
+    pub path: PathBuf,
+}
+
+/// Mapping describes how an individual path within the sandbox is connected to an external path
+/// in the underlying file system.
+#[derive(Debug, Eq, PartialEq)]
+pub struct Mapping {
+    path: PathBuf,
+    underlying_path: PathBuf,
+    writable: bool,
+}
+
+impl Mapping {
+    /// Creates a new mapping from the individual components.
+    ///
+    /// `path` is the inside the sandbox's mount point where the `underlying_path` is exposed.
+    /// Both must be absolute paths.
+    pub fn new(path: &Path, underlying_path: &Path, writable: bool)
```

Comment:
Every code path coerces these to PathBuf. Should you be asking the caller for PathBuf instead? It depends on the ergonomics.